vllm/v1/engine/llm_engine.py

```diff
@@ -142,18 +145,6 @@ def __init__(
         self.input_processor = input_registry.create_input_processor(
             model_config)
 
-        # Request id -> Request
-        self.requests: Dict[str, Request] = {}
-        # NOTE(woosuk): Now that the detokenizer works asynchronously, we need
-        # to keep track of how many steps each request has been lagged behind
-        # in terms of detokenization.
-        # Request id -> how many detokenizer steps the request should wait for.
-        self.num_lagged_steps: Dict[str, int] = {}
-        # OPTIMIZATION: Cache the request output and update it incrementally.
-        # This is used to avoid creating a new RequestOutput object every step.
-        # Request id -> RequestOutput
-        self.request_outputs: Dict[str, RequestOutput] = {}
```

Comment:
The key benefit of this PR is that we simplify the `LLMEngine` by no longer needing to keep track of these data.